added API reference (#102)
AdguardTeam/KnowledgeBaseDNS: docs/private-dns/api-reference.md

---
title: API Reference
sidebar_position: 3
toc_min_heading_level: 3
toc_max_heading_level: 4
---

<!-- 
    The content below is simply an automatic conversion from the OpenAPI spec
    https://api.adguard-dns.io/static/swagger/openapi.json to markdown using
    https://swagger-markdown-ui.netlify.app/.

    If you want to change it, ask the developers to change the OpenAPI spec.
-->

# AdGuard DNS API
DNS API documentation

> ## Version: 1.3

Reviewer: Update version to 1.4. API was updated recently (added custom blocking mode option) and I forgot to bump version in repo. Also maybe we need to add CHANGELOG here?

Author: It's autogenerated from https://api.adguard-dns.io/static/swagger/openapi.json

Author: Changelog is a good idea. Can you make it available on `api.adguard-dns.io`?

Reviewer: Done. New version at https://api.adguard-dns.io/static/swagger/openapi.json and [changelog](https://api.adguard-dns.io/static/api/CHANGELOG.md)

### /oapi/v1/account/limits

#### GET

> ##### Summary:

Reviewer 2: this subheading seems superfluous
probably, it's better to format the text in this way:

```
#### GET

 Get account limits

 ##### Responses

| Code | Description |
 | ---- | ----------- |
 | 200 | Account limits info |

#### POST

 Creates a new device

##### Responses
<...>
```

Author: This is auto-generated, let's keep it as is for the sake of simplicity.

> Get account limits

Reviewer 2: ```suggestion
Gets account limits
```

| ---- | ----------- |
| 200 | Device deleted |
| 404 | Device not found |

#### GET
##### Summary:

Gets an existing device by ID

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| device_id | path |  | Yes | string |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | Device info |
| 404 | Device not found |

#### PUT
##### Summary:

Updates an existing device

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| device_id | path |  | Yes | string |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | Device updated |
| 400 | Validation failed |
| 404 | Device not found |

### /oapi/v1/devices/{device_id}/doh.mobileconfig

#### GET
##### Summary:

Get DNS over HTTPS .mobileconfig.

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| device_id | path |  | Yes | string |
| exclude_wifi_networks | query | List Wi-Fi networks by their SSID, in which you want AdGuard DNS to be disabled | No | [ string ] |

> | exclude_domain | query | List domains that will use the default DNS servers instead of AdGuard DNS | No | [ string ] |

Reviewer 2: 'the' is not needed

> | exclude_domain | query | List domains that will use the default DNS servers instead of AdGuard DNS | No | [ string ] |

Author: Not necessarily, the default is the proper word here to be honest

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | DNS over HTTPS plist file |
| 404 | Device not found |

### /oapi/v1/devices/{device_id}/dot.mobileconfig

#### GET
##### Summary:

Get DNS over TLS .mobileconfig.

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| device_id | path |  | Yes | string |

> | exclude_wifi_networks | query | List Wi-Fi networks by their SSID, in which you want AdGuard DNS to be disabled | No | [ string ] |

Reviewer 2: extra comma before 'in which'

| exclude_domain | query | List domains that will use the default DNS servers instead of AdGuard DNS | No | [ string ] |

##### Responses

| Code | Description |
| ---- | ----------- |

> | 200 | DNS over HTTPS plist file |

Reviewer 2: DNS-over-HTTPS

| 404 | Device not found |

### /oapi/v1/devices/{device_id}/settings

#### PUT
##### Summary:

Updates device settings

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| device_id | path |  | Yes | string |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | Device settings updated |
| 400 | Validation failed |
| 404 | Device not found |

### /oapi/v1/dns_servers

#### GET
##### Summary:

Lists DNS servers

##### Description:

> Lists DNS servers for user. By default exists default DNS server

Reviewer 2: Lists DNS servers for a user

«по умолчанию существует дефолтный днс-сервер» — к чему это?

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | List of DNS servers |

#### POST
##### Summary:

Creates a new DNS server

##### Description:

> Creates a new DNS server. You can attach custom settings, otherwise DNS server will be created with default settings.

Reviewer 2: ```suggestion
Creates a new DNS serve that you can customize to your liking.
```
or, more formally:
```suggestion
Creates a new DNS serve that you can customize according to your preferences.
```

Author: Not really, the current description is more precise which is important for an API document.

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | DNS server created |
| 400 | Validation failed |
| 429 | DNS servers count reached the limit |

### /oapi/v1/dns_servers/{dns_server_id}

#### DELETE
##### Summary:

Removes a DNS server

##### Description:

> Removes a DNS server. All devices attached to this DNS server will be moved to the default DNS server. Deletion of default DNS server is forbidden.

Reviewer 2: Deletion of —> Deleting

Gets an existing DNS server by ID

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| dns_server_id | path |  | Yes | string |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | DNS server info |
| 404 | DNS server not found |

#### PUT
##### Summary:

Updates an existing DNS server

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| dns_server_id | path |  | Yes | string |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | DNS server updated |
| 400 | Validation failed |
| 404 | DNS server not found |

### /oapi/v1/dns_servers/{dns_server_id}/settings

#### PUT
##### Summary:

Updates DNS server settings

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| dns_server_id | path |  | Yes | string |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | DNS server settings updated |
| 400 | Validation failed |
| 404 | DNS server not found |

### /oapi/v1/filter_lists

#### GET
##### Summary:

> Get filter lists

Reviewer 2: Gets

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | List of filters |

### /oapi/v1/oauth_token

#### POST
##### Summary:

> Generate Access and Refresh token

Reviewer 2: Generates

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | Access token issued |
| 400 | Missing required parameters |
| 401 | Invalid credentials, MFA token or refresh token provided |

null

### /oapi/v1/query_log

#### DELETE
##### Summary:

> Clear query log

Reviewer 2: Clears

##### Responses

| Code | Description |
| ---- | ----------- |
| 202 | Query log was cleared |

#### GET
##### Summary:

> Get query log

Reviewer 2: Gets

| time_to_millis | query | Time to in milliseconds (inclusive) | Yes | long |
| devices | query | Filter by devices | No | [ string ] |
| countries | query | Filter by countries | No | [ string ] |
| companies | query | Filter by companies | No | [ string ] |
| statuses | query | Filter by statuses | No | [ [FilteringActionStatus](#FilteringActionStatus) ] |
| categories | query | Filter by categories | No | [ [CategoryType](#CategoryType) ] |
| search | query | Filter by domain name | No | string |
| limit | query | Limit the number of records to be returned | No | integer |
| cursor | query | Pagination cursor. Use cursor from response to paginate through the pages. | No | string |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | Query log |

### /oapi/v1/revoke_token

#### POST
##### Summary:

Revokes a Refresh Token

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| refresh_token | query | Refresh Token | Yes | string |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | Refresh token revoked |

null

### /oapi/v1/stats/categories

#### GET
##### Summary:

> Get categories statistics

Reviewer 2: Gets

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| time_from_millis | query | Time from in milliseconds (inclusive) | Yes | long |
| time_to_millis | query | Time to in milliseconds (inclusive) | Yes | long |
| devices | query | Filter by devices | No | [ string ] |
| countries | query | Filter by countries | No | [ string ] |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | Categories statistics received |
| 400 | Validation failed |

### /oapi/v1/stats/companies

#### GET
##### Summary:

> Get companies statistics

Reviewer 2: Gets

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| time_from_millis | query | Time from in milliseconds (inclusive) | Yes | long |
| time_to_millis | query | Time to in milliseconds (inclusive) | Yes | long |
| devices | query | Filter by devices | No | [ string ] |
| countries | query | Filter by countries | No | [ string ] |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | Companies statistics received |
| 400 | Validation failed |

### /oapi/v1/stats/companies/detailed

#### GET
##### Summary:

> Get detailed companies statistics

Reviewer 2: Gets

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| time_from_millis | query | Time from in milliseconds (inclusive) | Yes | long |
| time_to_millis | query | Time to in milliseconds (inclusive) | Yes | long |
| devices | query | Filter by devices | No | [ string ] |
| countries | query | Filter by countries | No | [ string ] |
| cursor | query | Pagination cursor | No | string |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | Detailed companies statistics received |
| 400 | Validation failed |

### /oapi/v1/stats/countries

#### GET
##### Summary:

> Get countries statistics

Reviewer 2: Gets

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| time_from_millis | query | Time from in milliseconds (inclusive) | Yes | long |
| time_to_millis | query | Time to in milliseconds (inclusive) | Yes | long |
| devices | query | Filter by devices | No | [ string ] |
| countries | query | Filter by countries | No | [ string ] |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | Countries statistics received |
| 400 | Validation failed |

### /oapi/v1/stats/devices

#### GET
##### Summary:

> Get devices statistics

Reviewer 2: Gets

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| time_from_millis | query | Time from in milliseconds (inclusive) | Yes | long |
| time_to_millis | query | Time to in milliseconds (inclusive) | Yes | long |
| devices | query | Filter by devices | No | [ string ] |
| countries | query | Filter by countries | No | [ string ] |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | Devices statistics received |
| 400 | Validation failed |

### /oapi/v1/stats/domains

#### GET
##### Summary:

> Get domains statistics

Reviewer 2: Gets

##### Parameters

| Name | Located in | Description | Required | Schema |
| ---- | ---------- | ----------- | -------- | ---- |
| time_from_millis | query | Time from in milliseconds (inclusive) | Yes | long |
| time_to_millis | query | Time to in milliseconds (inclusive) | Yes | long |
| devices | query | Filter by devices | No | [ string ] |
| countries | query | Filter by countries | No | [ string ] |

##### Responses

| Code | Description |
| ---- | ----------- |
| 200 | Domains statistics received |
| 400 | Validation failed |

### /oapi/v1/stats/time

#### GET
##### Summary:

> Get time statistics

Reviewer 2: Gets
